src/transaction/log_page_buffer.c

```diff
@@ -2339,27 +2439,39 @@ logpb_read_page_from_file_or_page_server (THREAD_ENTRY * thread_p, LOG_PAGEID pa
   const SERVER_TYPE server_type = get_server_type ();
   if (server_type == SERVER_TYPE_TRANSACTION && ts_Gl->is_page_server_connected ())
     {
+      // *INDENT-OFF*
+      auto page_request_func = [pageid](LOG_PAGE* log_pgptr) -> int
```

Comment:
I do do understand why this indirection is still needed.

Do requests still come for `LOGPB_HEADER_PAGE_ID` via `logpb_read_page_from_file_or_page_server` function ?

There are still 2 calls to
```
    logpb_read_page_from_file_or_page_server (thread_p, LOGPB_HEADER_PAGE_ID, LOG_CS_FORCE_USE, temp_pgptr);
```
in the function `log_create_internal` which I have added some time back.
The calls are guarded by `#if defined(CUBRID_DEBUG)` - thus, not executed by default.
I now think that those are incorrect and should be replaced back to `logpb_read_page_from_file` because the function `log_create_internal` should not be called on a non-PS?
I did not debug to check this.

Reply:
I run into some cases the HDR log page is retrieved with this function. For example, 
`logpb_fetch_header-> logpb_fetch_header_with_buffer -> ... -> logpb_read_page_from_file_or_page_server`.
 And, I just fixed this function to handle the case naively. However, yes, if a HDR page is requested it should come through `logpb_hdr_read_page_from_file_or_page_server`, which is for the case. I should've, but I missed.

Having analyzed further, there are two cases for getting the HDR log page on the 'monolithic' server:
- On Running: `logpb_fetch_header_with_buffer`, which gets it through the general log page read interface, `logpb_fetch_page`.
- On Recovery: `logpb_fetch_header_from_file`

With the modification in this PR, the `logpb_hdr_read_page_from_file_or_page_server` uses `logpb_fetch_header_from_file`. It can't be used for the `logpb_fetch_header` case. I should come up with a new idea taking this into account.

The `logpb_fetch_header_with_buffer` is used in a few not general case, but I think the HDR page should be handled like a normal log page while running. I'm not sure, though. I'll get back with a new solution.